### PR TITLE
Binary Operations on DataFrame

### PR DESCRIPTION
**Pull request by Kriyszig**
Issue: #1 

New ways to construct index, declare DataFrame and Bianry Operations on DataFrame Rows and Columns.

### CODE REVIEW

File: source/magpie/dataframe.d [hunk 1]
```diff
@@ -1023,6 +1024,9 @@ public:
     auto opIndex(Args...)(Args args)
         if(Args.length > 0 && Args.length < 3)
     {
+        static assert(is(Args[0] == string[]), "Indexes must be an array of string");
+        static if(Args.length > 1)
+            static assert(is(Args[1] == int), "Mention axis as 0 if you want to fetch a row");
```
Comment (thewilsonator) [inline source/magpie/dataframe.d]: That message could use some rewording.

File: source/magpie/dataframe.d [hunk 2]
```diff
@@ -1048,9 +1056,47 @@ public:
 
             return retrow;
         }
-
+
```
Comment (thewilsonator) [inline source/magpie/dataframe.d]: Its not great to add bits of whitespace, although its doesn't really matter.

File: source/magpie/dataframe.d (hunk 1 above)
Comment (Kriyszig) [reply source/magpie/dataframe.d]: Addressed in cbe18d9
Improved the message explaining the correct usage for binary operations.

File: source/magpie/dataframe.d (hunk 2 above)
Comment (Kriyszig) [reply source/magpie/dataframe.d]: Apologies. Sometimes the cursor jumps on due to poor palm rejection on my laptom :sweat_smile: 
Fixed in cbe18d9 - addressed the complete linting issue

**Pull request merged**